File: backend/app/core/currencies.py
"""Fixed reference list of ISO 4217 currency codes for donations.currency.

Ordered as the club wants it presented: the most-used currencies pinned to
the top (HKD, SGD, USD, EUR, JPY, CNY), then the rest of the common codes
alphabetically. Store the ISO code (e.g. CNY, not "RMB") in the database.
Keep this in sync with frontend/src/data/currencies.js — there is no shared
package between the two apps, so the list is intentionally duplicated.
"""

PINNED_CURRENCIES: tuple[str, ...] = ("HKD", "SGD", "USD", "EUR", "JPY", "CNY")

OTHER_CURRENCIES: tuple[str, ...] = (
    "AED", "ARS", "AUD", "BHD", "BRL", "CAD", "CHF", "CZK",
    "DKK", "GBP", "HUF", "IDR", "ILS", "INR", "KRW", "KWD",
    "MOP", "MXN", "MYR", "NOK", "NZD", "PHP", "PLN", "QAR",
    "RON", "RUB", "SAR", "SEK", "THB", "TRY", "TWD", "VND",
    "ZAR",
)

CURRENCIES: tuple[str, ...] = PINNED_CURRENCIES + OTHER_CURRENCIES

File: frontend/src/data/currencies.js
// Fixed reference list of ISO 4217 currency codes for the donation currency
// dropdown. Pinned currencies (most-used by the club) come first, then the
// rest alphabetically. Mirrors backend/app/core/currencies.py — there is no
// shared package between the two apps, so the list is intentionally
// duplicated. "RMB" is the common name for CNY; we store/display the ISO
// code but label it "CNY (RMB)" for clarity.
export const PINNED_CURRENCIES = ["HKD", "SGD", "USD", "EUR", "JPY", "CNY"];

export const OTHER_CURRENCIES = [
  "AED", "ARS", "AUD", "BHD", "BRL", "CAD", "CHF", "CZK",
  "DKK", "GBP", "HUF", "IDR", "ILS", "INR", "KRW", "KWD",
  "MOP", "MXN", "MYR", "NOK", "NZD", "PHP", "PLN", "QAR",
  "RON", "RUB", "SAR", "SEK", "THB", "TRY", "TWD", "VND",
  "ZAR",
];

export const CURRENCIES = [...PINNED_CURRENCIES, ...OTHER_CURRENCIES];

const LABEL_OVERRIDES = { CNY: "CNY (RMB)" };

export function currencyLabel(code) {
  return LABEL_OVERRIDES[code] ?? code;
}
